test/linting/lint.sh

```diff
@@ -104,73 +90,41 @@ function _hadolint
 function _shellcheck
 {
   local SCRIPT='SHELLCHECK'
-  local ERR=0
-  local LINT=(shellcheck -x -S style -Cauto -o all -e SC2154 -W 50)
-
-  if ! __in_path "${LINT[0]}"
-  then
-    __log_failure 'linter not in PATH'
-    return 2
-  fi
-
-  __log_info "$(${LINT[0]} --version | grep -m 2 -o "[0-9.]*")"
-
-  # an overengineered solution to allow shellcheck -x to
-  # properly follow `source=<SOURCE FILE>` when sourcing
-  # files with `. <FILE>` in shell scripts.
-  while read -r FILE
-  do
-    if ! (
-      cd "$(realpath "$(dirname "$(readlink -f "${FILE}")")")"
-      if ! "${LINT[@]}" "$(basename -- "${FILE}")"
-      then
-        exit 1
-      fi
-    )
-    then
-      ERR=1
-    fi
-  done < <(find . -type f -iname "*.sh" \
-    -not -path "./test/bats/*" \
-    -not -path "./test/test_helper/*" \
-    -not -path "./target/docker-configomat/*")
-
-  # the same for executables in target/bin/
-  while read -r FILE
-  do
-    if ! (
-      cd "$(realpath "$(dirname "$(readlink -f "${FILE}")")")"
-      if ! "${LINT[@]}" "$(basename -- "${FILE}")"
-      then
-        exit 1
-      fi
-    )
-    then
-      ERR=1
-    fi
-  done < <(find target/bin -executable -type f)
-
-  # the same for all test files
-  while read -r FILE
-  do
-    if ! (
-      cd "$(realpath "$(dirname "$(readlink -f "${FILE}")")")"
-      if ! "${LINT[@]}" "$(basename -- "${FILE}")"
-      then
-        exit 1
-      fi
-    )
-    then
-      ERR=1
-    fi
-  done < <(find test/ -maxdepth 1 -type f -iname "*.bats")
-
-  if [[ ${ERR} -eq 1 ]]
+  
+  # File paths for shellcheck:
+  F_SH="$(find . -type f -iname '*.sh' \
+    -not -path './test/bats/*' \
+    -not -path './test/test_helper/*' \
+    -not -path './target/docker-configomat/*' -print0 \
+    | xargs -0
+  )"
+  [[ "$(uname)" == "Darwin" ]] && PERM="+111" || PERM="/111"
+  F_BIN="$(find 'target/bin' -perm "${PERM}" -type f -or -type l)"
```

Comment:
This is only a suggestion if the other maintainers would prefer it. Since both variants produce 21 files, there is no parity concern at present:

> _It took me a while to understand the implications of "not identical to the `-executable` predicate" and "just tests if any execute permissions are set": It means that `-perm +111` **may yield false positives**, i.e., files that the current user cannot actually execute._
> _There's no way to emulate `-executable` by testing permissions alone, because what's needed is to relate the file's user and group identity to the current user's._

I don't see why we'd ever run into that difference, so this suggestion probably doesn't need to be applied.

```suggestion
  # macOS lacks parity for `-executable` but presently produces the same results: https://stackoverflow.com/a/4458361
  [[ "$(uname)" == "Darwin" ]] && FIND_EXEC="-perm +111 -type l -or" || FIND_EXEC="-executable"
  F_BIN="$(find 'target/bin' "${FIND_EXEC}" -type f)"
```